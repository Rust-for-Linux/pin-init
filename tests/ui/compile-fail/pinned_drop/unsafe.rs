use pinned_init::*;
use std::pin::Pin;

#[pin_data(PinnedDrop)]
struct Foo {}

#[pinned_drop]
unsafe impl PinnedDrop for Foo {
    fn drop(self: Pin<&mut Self>) {}
}

fn main() {}
