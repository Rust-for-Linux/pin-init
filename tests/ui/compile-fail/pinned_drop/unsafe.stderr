error: custom attribute panicked
 --> tests/ui/compile-fail/pinned_drop/unsafe.rs:7:1
  |
7 | #[pinned_drop]
  | ^^^^^^^^^^^^^^
  |
  = help: message: assertion failed: matches!(&toks[0], TokenTree::Ident(i) if *i == "impl")

warning: unused import: `std::pin::Pin`
 --> tests/ui/compile-fail/pinned_drop/unsafe.rs:2:5
  |
2 | use std::pin::Pin;
  |     ^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0277]: the trait bound `Foo: PinnedDrop` is not satisfied
 --> tests/ui/compile-fail/pinned_drop/unsafe.rs:4:1
  |
4 | #[pin_data(PinnedDrop)]
  | ^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | the trait `PinnedDrop` is not implemented for `Foo`
  | required by a bound introduced by this call
  |
  = note: this error originates in the macro `$crate::__pin_data` which comes from the expansion of the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)
